diff --git a/node_modules/react-quill/lib/index.js b/node_modules/react-quill/lib/index.js
index 733857e..b3252a1 100644
--- a/node_modules/react-quill/lib/index.js
+++ b/node_modules/react-quill/lib/index.js
@@ -1026,7 +1026,7 @@
 	 *
 	 */
 	ReactQuill.prototype.getEditingArea = function getEditingArea() {
-		return _reactDom2.default.findDOMNode(this.editingArea);
+		return this.editingArea.current;
 	};
 
 	/**
@@ -1183,12 +1183,13 @@
 		key: 'render',
 		value: function render() {
 			var _this2 = this;
+			this.editingArea = _react2.default.createRef();
 
 			return _react2.default.createElement(
 				'div',
 				{
 					id: this.props.id,
-					style: this.props.style,
+					style: this.props.style,					
 					className: ['quill'].concat(_toConsumableArray(this.props.className.split(' '))).join(' '),
 					onKeyPress: this.props.onKeyPress,
 					onKeyDown: this.props.onKeyDown,
@@ -1197,9 +1198,7 @@
 						_this2.state.generation,
 					),
 					dangerouslySetInnerHTML: { __html: this.getHTML() },
-					ref: function ref(node) {
-						_this2.editingArea = node;
-					},
+					ref: this.editingArea,
 					onMouseDown: this.onEditorMouseDown
 				})
 			);
